### PR TITLE
Fixed all PHPStan errors.

### PR DESCRIPTION
**Pull request by bariscodefxy**
You can review my additions and deletions.

### CODE REVIEW

File: src/Graphics/Rendering/Renderer/Debug3DRenderer.php [hunk 1]
```diff
@@ -299,12 +299,13 @@ public function addRay(
...
-     * @param Vec3 $destination The destination of the line
-     * @param Vec3 $color The color of the line
+     *
+     * @param Vec3|float $origin
+     * @param Vec3|float $destination
+     * @param Vec3|float $color
+     * @return void
      */
-    public function addLine(Vec3 $origin, Vec3 $destination, Vec3 $color) : void
+    public function addLine(Vec3|float $origin, Vec3|float $destination, Vec3|float $color) : void
```
Comment (mario-deluna) [inline src/Graphics/Rendering/Renderer/Debug3DRenderer.php]: Cannot be float, is always `Vec3`

File: src/Geo/AABB.php [hunk 2]
```diff
@@ -58,7 +58,7 @@ public function copy() : AABB
     /**
      * Returns the center of the AABB
      */
-    public function getCenter() : Vec3
+    public function getCenter() : float|Vec3
```
Comment (mario-deluna) [inline src/Geo/AABB.php]: Is always `Vec3` phpstan unfortunately is no able to handle the operator overloading properly. 

But Vec3 * float = Vec3

File: src/Geo/AABB2D.php [hunk 3]
```diff
@@ -70,7 +70,7 @@ public function copy() : AABB2D
     /**
      * Returns the center of the AABB
      */
-    public function getCenter() : Vec2
+    public function getCenter() : float|Vec2
```
Comment (mario-deluna) [inline src/Geo/AABB2D.php]: Same here

**Review by mario-deluna — CHANGES_REQUESTED** [review]
Thanks! There are 3 small type issues if they are resolved I very gladly merge this.

File: src/Geo/AABB2D.php (hunk 3 above)
Comment (bariscodefxy) [reply src/Geo/AABB2D.php]: it's not only float, it's float or Vec3

Comment (mario-deluna) [reply src/Geo/AABB2D.php]: Yes, but that is incorrect, look at the source: https://github.com/mario-deluna/php-glfw/blob/master/phpglfw_math.c#L317

This is the implementation of the `getCenter`function:

```
($this->min + $this->max) * 0.5;
```

aka 

```
(Vec2 + Vec2) * 0.5 
```

This expression will **always** return a `Vec2` and **never** a scalar `float`.

Comment (bariscodefxy) [reply src/Geo/AABB2D.php]: So, should I remove `float` return types, then add these to `phpstan.neon` as ignored errors?

**Comment by bariscodefxy** [comment]
I think this is outdated and useless.

**Pull request closed without merging**